File: src/main/java/me/miki/shindo/utils/file/filter/sound/WavFileFilter.java
package me.miki.shindo.utils.file.filter.sound;

import me.miki.shindo.utils.file.FileUtils;

import javax.swing.filechooser.FileFilter;
import java.io.File;

public class WavFileFilter extends FileFilter {

    @Override
    public boolean accept(File file) {

        if (file.isDirectory()) {
            return true;
        }

        String extension = FileUtils.getExtension(file);

        return extension != null && extension.equalsIgnoreCase("wav");
    }

    @Override
    public String getDescription() {
        return "Sounds (*.wav)";
    }
}

File: src/main/kotlin/com/shindoclient/shindo/utils/file/filter/sound/WavFileFilter.kt
package com.shindoclient.shindo.utils.file.filter.sound

import com.shindoclient.shindo.utils.file.FileUtils
import java.io.File
import javax.swing.filechooser.FileFilter

class WavFileFilter : FileFilter() {
    override fun accept(file: File?): Boolean {
        if (file == null) return false
        if (file.isDirectory) return true
        val extension = FileUtils.getExtension(file)
        return extension != null && extension.equals("wav", ignoreCase = true)
    }

    override fun getDescription(): String = "WAV Audio (*.wav)"
}
